### ISSUE TITLE
`react/hook-use-state` false positive "not destructured into value" error

### ISSUE THREAD

**Issue by EvHaus** [yannickcr/eslint-plugin-react#3224]
The following code is triggering this error:

```
react/hook-use-state: useState call is not destructured into value + setter pair
```

The code:

```js
import {useState} from 'react';

export const MyComponent = () => {
	const [isModalOpen, setIsModelOpen] = useState(false);
	return null;
};

export default MyComponent;
```

I think this is a false positive trigger because the code follows all the rules described in the rule. I suspect there's something that's not right about the name of those variables but I can't figure out exactly what's wrong.

CC @duncanbeevers

**Comment by EvHaus** [comment]
🤦 I have a typo: "modal" vs "model". Nevermind.

**Comment by duncanbeevers** [comment]
@EvHaus You should also have been able to use the suggested fix. Did your IDE offer you the suggestion?
